src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/sentiment/SentimentDetector.scala

```diff
@@ -1,49 +1,31 @@
 package com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment
 
-import net.liftweb.json
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
 
-import scalaj.http.Http
-
-case class SentimentDetectorAuth(key: String, apiHost: String = "westus.api.cognitive.microsoft.com")
+import scala.util.{Failure, Success, Try}
 
 @SerialVersionUID(100L)
 class SentimentDetector(
   auth: SentimentDetectorAuth
-) extends Serializable {
-
-  def detectSentiment(text: String, language: String): Option[Double] = {
-    val textId = "0"
-    val requestBody = buildRequestBody(text, textId, language)
-    val response = callCognitiveServices(requestBody)
-    parseResponse(response, textId)
-  }
-
-  protected def callCognitiveServices(requestBody: String): String = {
-    Http(s"https://${auth.apiHost}/text/analytics/v2.0/sentiment")
-      .headers(
-        "Content-Type" -> "application/json",
-        "Ocp-Apim-Subscription-Key" -> auth.key)
-      .postData(requestBody)
-      .asString
-      .body
-  }
+) extends Serializable with Loggable {
 
-  protected def buildRequestBody(text: String, textId: String, language: String): String = {
-    implicit val formats = json.DefaultFormats
-    val requestBody = dto.JsonSentimentDetectionRequest(documents = List(dto.JsonSentimentDetectionRequestItem(
-      id = textId,
-      language = language,
-      text = text)))
-    json.compactRender(json.Extraction.decompose(requestBody))
-  }
+  private lazy val cognitiveServicesSentimentDetector = new CognitiveServicesSentimentDetector(auth)
+  private lazy val wordlistSentimentDetector = new WordListSentimentDetector()
 
-  protected def parseResponse(apiResponse: String, textId: String): Option[Double] = {
-    implicit val formats = json.DefaultFormats
-    val response = json.parse(apiResponse).extract[dto.JsonSentimentDetectionResponse]
-    if (response.errors.exists(_.id == textId)) {
-      None
-    } else {
-      response.documents.find(_.id == textId).map(_.score)
+  def detectSentiment(text: String, language: String): Option[Double] = {
+    val sentiment = Try(cognitiveServicesSentimentDetector.detectSentiment(text, language))
```

Comment:
I think we should do this now since this is new code. We're doing error handling in other components (stream provider, for example).